### PR TITLE
🌱 Deprecate the HealthCheckSucceded condition for typo

### PR DESCRIPTION
**Pull request by qmloong**
**What this PR does / why we need it**:
Fix typo of the `HealthCheckSucceded` condition

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
N/A

### CODE REVIEW

**Comment by enxebre** [comment]
Thanks!

/ok-to-test
/lgtm

**Review by killianmuldoon — APPROVED** [review]
/lgtm

File: api/v1beta1/condition_consts.go [hunk 1]
```diff
@@ -132,8 +132,13 @@ const (
 )
 
 const (
+	// MachineHealthCheckSucceededCondition is set on machines that have passed a healthcheck by the MachineHealthCheck controller.
+	// In the event that the health check fails it will be set to False.
+	MachineHealthCheckSucceededCondition ConditionType = "HealthCheckSucceeded"
+
 	// MachineHealthCheckSuccededCondition is set on machines that have passed a healthcheck by the MachineHealthCheck controller.
 	// In the event that the health check fails it will be set to False.
+	// Deprecated: This const is going to be removed in a next release.
```
Comment (fabriziopandini) [inline api/v1beta1/condition_consts.go:141]: nit
```suggestion
	// Deprecated: This const is going to be removed in a next release. Use MachineHealthCheckSucceededCondition instead.
```

**Review by fabriziopandini — COMMENTED** [review]
only a nit, other wise lgtm

File: api/v1beta1/condition_consts.go (hunk 1 above)
Comment (qmloong) [reply api/v1beta1/condition_consts.go:141]: Got it. I will update it later.

**Comment by enxebre** [comment]
changes looks good @qmloong, please squash into a single commit

**Comment by qmloong** [comment]
> changes looks good @qmloong, please squash into a single commit

Done

**Pull request merged**